harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -556,7 +559,12 @@ def _compute_validation_metrics(self) -> workload.Response:
                 metrics_reducers=self._prepare_metrics_reducers(keys=keys),
             )
 
-            num_inputs *= self.context.distributed.size
+            # Gather a list of per-worker (num_inputs, num_batches) tuples.
+            input_counts = self.context.distributed._zmq_gather((num_inputs, idx + 1))
```

Comment:
note: i think we'll actually run into a hang if dataloader lengths are not the same across slots so idx should be the same.

Reply:
I don't think that's the case, I think validation is all done asynchronously on each shard and metrics are reduced after all are finished.

Reply:
> I don't think that's the case, I think validation is all done asynchronously on each shard and metrics are reduced after all are finished.

Yep, I see.  It is okay if model forward calls have no synchronized distributed communication but blocking if there is something like sync batch norm.